07-data_pipelines.Rmd

```diff
@@ -157,3 +157,74 @@ Link: https://airflow.tools.alpha.mojanalytics.xyz
 You can manually start ("trigger") the pipeline with this button:
 
 ![](images/pipelines/trigger.png)
+
+
+## Test pipeline in your own Airflow sandbox
+
+You can deploy your own sandboxed version of Airflow to test Dags and docker images on.
+
+### Start Airflow sandbox
+
+Click the `Deploy` button next to `airflow-sqlite` in the Analytical tools section of the control panel. This should take a few minutes to deploy. Then Click the `Open` button to open your Airflow sandbox in a new window.
+
+This will creat an `airflow` directory in your home directory on the Analytical Platform. In this airflow directory there are three more directories `db`, `dags` and `logs`. The `dags` directory is where you will store your test dag files.
+
+### Create a test Dag
+
+Using Jupiter lab on the Analytical Platform create a python file in the `airflow/dags` directory in your home directory in the Analytical Platform (eg. `test_dag.py`). Airflow will automatically scan this directory for dag files every three minutes.
+
+See above for creating a Docker image to run the dag if you are using `KubernetesPodOperator`. Concourse will automatically build images of PR's to github airflow repos and will be tagged with the name of the PR.
+
+Example Dag file contents:
+
+
+    from airflow.utils.dates import days_ago
+    from airflow.utils.log.logging_mixin import LoggingMixin
+    from airflow.models import DAG
+
+    log = LoggingMixin().log
+
+    try:
+        from airflow.contrib.operators.kubernetes_pod_operator import KubernetesPodOperator
+
+        args = {"owner": "airflow", "start_date": days_ago(2)}
+
+        dag = DAG(
+            dag_id="example_kubernetes_operator_assume_role",
+            default_args=args,
+            schedule_interval=None,
+        )
+
+        k = KubernetesPodOperator(
+            namespace="user-###YOUR_GITHUB_USERNAME###",
+            image="governmentpaas/awscli:latest",
+            cmds=["aws", "--debug"],
+            arguments=["s3api", "list-objects", "--bucket", "###YOUR_TEST_BUCKET###"],
+            env_vars={
+                "AWS_METADATA_SERVICE_TIMEOUT": "60",
+                "AWS_METADATA_SERVICE_NUM_ATTEMPTS": "5",
+            },
+            labels={"foo": "bar"},
+            name="airflow-test-pod",
+            in_cluster=True,
+            task_id="task",
+            get_logs=True,
+            dag=dag,
+            annotations={"iam.amazonaws.com/role": "alpha_user_###YOUR_GITHUB_USERNAME###"},
+        )
+
+
+    except ImportError as e:
+        log.warn("Could not import KubernetesPodOperator: " + str(e))
+
+
+
+You will have to set the dag namespace to your namespace and the aws role to your role, see `###YOUR_GITHUB_USERNAME###` in the above example code.
+
+You will also need to update the image with your dag image from AWS ECR and add any environment variables etc. as you would with a normal Dag.
+
+### Caveats
```

Comment:
I'd probably change this to **IMPORTANT** as we want to be very sure users will not start using this to run jobs as an alternative to the main Airflow.